src/game/UILevelGuide.cpp

```diff
@@ -236,10 +120,140 @@ bool CUILevelGuide::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
             return true;
         }
 
-        if (m_pText_Guides[iIndex]->GetString().length() > 0) {
-            m_pText_Guides[iIndex]->SetStartLine(m_pScroll_Guides[iIndex]->GetCurrentPos());
-        }
+        m_pText_Guides[iIndex]->SetStartLine(m_pScroll_Guides[iIndex]->GetCurrentPos());
     }
 
     return true;
 }
+
+void CUILevelGuide::SetVisible(bool bVisible) {
+    if (m_bVisible == bVisible) {
+        return;
+    }
+
+    if (bVisible) {
+        UpdateQuestsByLevel(CGameBase::s_pPlayer->m_InfoBase.iLevel);
+        if (m_pEdit_Level) {
+            m_pEdit_Level->SetString("");
+            m_pEdit_Level->SetFocus();
+        }
+    } else {
+        if (m_pEdit_Level) {
+            m_pEdit_Level->SetString("");
+            m_pEdit_Level->KillFocus();
+        }
+    }
+
+    CN3UIBase::SetVisible(bVisible);
+    if (bVisible) {
+        CGameProcedure::s_pUIMgr->SetVisible(this);
+    } else {
+        CGameProcedure::s_pUIMgr->ReFocusUI();
+    }
+}
+
+void CUILevelGuide::UpdateQuestsByLevel(int iLevel) {
+    if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + 5 >= iLevel) {
```

Comment:
change it to 

 if (iLevel >= 0 && CGameBase::s_pPlayer->m_InfoBase.iLevel + MAX_SEARCH_ABOVE_LEVEL >= iLevel) {